python/pyarrow/parquet.py

```diff
@@ -1252,6 +1355,95 @@ def _make_manifest(path_or_paths, fs, pathsep='/', metadata_nthreads=1,
     return pieces, partitions, common_metadata_path, metadata_path
 
 
+class _ParquetDatasetV2:
+    """
+    ParquetDataset shim using the Dataset API under the hood.
+    """
+    def __init__(self, path_or_paths, filesystem=None, filters=None,
+                 partitioning="hive", read_dictionary=None, buffer_size=None,
+                 memory_map=False, **kwargs):
+        import pyarrow.dataset as ds
+        import pyarrow.fs
+
+        # Raise error for not supported keywords
+        for keyword, default in [
+                ("schema", None), ("metadata", None),
+                ("split_row_groups", False), ("validate_schema", True),
+                ("metadata_nthreads", 1)]:
+            if keyword in kwargs and kwargs[keyword] is not default:
+                raise ValueError(
+                    "Keyword '{0}' is not yet supported with the new "
+                    "Dataset API".format(keyword))
+
+        # map old filesystems to new one
+        # TODO(dataset) deal with other file systems
```

Comment:
Would it be hard to map HadoopFileSystem and S3FSWrapper as well?
Either way please file a jira ticket for it.

Reply:
The question might also be if we actually want to add such mapping (for LocalFileSystem I mainly did it to get the tests passing right now). 
As in the end, we also want that people switch to the new filesystems, so maybe we should rather give that message (certainly if they do the `use_legacy_dataset=False` opt-in, they can also use the new filesystem objects?)